Mono.Addins/Mono.Addins/ExtensionNode.cs

```diff
@@ -459,8 +459,21 @@ AddinLocalizer GetAddinLocalizer ()
 				return Addin.Localizer;
 
 			Addin foundAddin = addinEngine.Registry.GetAddin (addinId);
-			if (foundAddin == null || foundAddin.Description.Localizer != null)
+			if (foundAddin == null || foundAddin.Description.Localizer != null) {
+				// Try checking if there's any id for a reusable localizer without having to load the addin.
+
+				var localizerType = addinEngine.GetLocalizerType (foundAddin.Description.LocalId);
```

Comment:
This doesn't look right. It should use the id of the localizer, not the id of the add-in.

Reply:
Good catch. There's some stuff I moved around and it seems like I screwed it up!